homeassistant/components/suez_water/coordinator.py

```diff
@@ -37,10 +63,22 @@ async def _async_setup(self) -> None:
         if not await self._suez_client.check_credentials():
             raise ConfigEntryError("Invalid credentials for suez water")
 
-    async def _async_update_data(self) -> AggregatedData:
+    async def _async_update_data(self) -> SuezWaterData:
         """Fetch data from API endpoint."""
         try:
-            data = await self._suez_client.fetch_aggregated_data()
+            aggregated = await self._suez_client.fetch_aggregated_data()
+            data = SuezWaterData(
+                aggregated_value=aggregated.value,
+                aggregated_attr={
+                    "this_month_consumption": aggregated.current_month,
+                    "previous_month_consumption": aggregated.previous_month,
+                    "highest_monthly_consumption": aggregated.highest_monthly_consumption,
+                    "last_year_overall": aggregated.previous_year,
+                    "this_year_overall": aggregated.current_year,
+                    "history": aggregated.history,
```

Comment:
Shouldn't these become sensors of their own instead of attributes?

Reply:
Already mentioned that in a previous review :)

Reply:
yes they should, but I have previously with @joostlek to do the changes incrementally.
Plus if I do it, it will be breaking change for existing users and I don't know how this should be handled.

And the next change I'm planning is to fetch directly in recorder statistics so it can used in energy dashboard, and in doing so all (most) of these data will become obsolete.